feature/home/src/main/java/com/yapp/home/util/AlarmDateTimeFormatter.kt

```
@@ -0,0 +1,185 @@
+
+    fun getFormattedEarliestUpcomingAlarmDeliveryTime(
+        alarms: List<Alarm>,
+        formats: DeliveryTimeFormats,
+        now: LocalDateTime = LocalDateTime.now(clock),
+    ): String {
+        val earliestAlarmDateTime = alarms
+            .filter { it.isAlarmActive }
+            .mapNotNull { alarm ->
+                try {
+                    calculateNextOccurrence(alarm.hour, alarm.minute, alarm.repeatDays, now)
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)